### DISCUSSION TITLE
HSV and related color spaces

### DISCUSSION THREAD

**Discussion by claus-01** [ImageMagick/ImageMagick#2318] [General] [answered]
Hi,

is there an easy to understand explanation on the difference between HSV, HSL, HSI, and HSB? Why four different color spaces?

TIA,

Claus

**Comment by snibgo** [comment, marked as answer]
HSB and HSV are the same. B = V = max(R,G,B)

HSL: L = min(R,G,B)/2+max(R,G,B)/2

HSI: I = (R+G+B)/3 [EDIT: corrected from 2 to 3.]

Why are there three versions? They were developed historically for various purposes, eg for selection in colour-wheels.

I think of these as _graphic_ colorspaces rather than _photographic_ colorspaces. They treat red, green and blue equivalently. For example, pure red, green or blue has the same values of B, V, L or I. Similarly pure yellow and pure magenta.

**Reply by claus-01** [reply]
Is min(R,G,B) black & max(R,G,B) white? And: Where's the difference between (R+G+B) and (R, G,B)? Could you possibly insert numbers (like 255 or FF) to illustrate this example?

Thanks,

Claus

**Comment by snibgo** [comment]
Sorry, I messed up the HSI formula, now corrected.

I'll use example values on a scale of 0 to 255. Suppose RGB=(10,20,30), so the red channel is 10, green is 20 and blue is 30, so this colour is a dull, dark blue.

max(10,20,30) means the maximum (ie largest) of the three channels, so this is 30 (also on a scale of 0 to 255).

min(10,20,30) is 10.

min(R,G,B)/2+max(R,G,B)/2 = 10/2 + 30/2 = 5 + 15 = 20.

(R+G+B)/2 = (10+20+30)/3 = 60/3 = 20.
